### PR TITLE
Update to use node 12 typings

### LINKED ISSUE #82514: Dependencies: update to node @types 12.x
When we ship with Electron 6, we should update our `@types/node` dependency to 12.x as well for:
* [ ] vscode core
* [x] all our built in extensions

I would wait before updating `build/package.json` until we do https://github.com/microsoft/vscode/issues/80842

### PR DESCRIPTION
**Pull request by mjbvz**
Fixes #82514

### CODE REVIEW

File: src/vs/base/common/async.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export function createCancelablePromise<T>(callback: (token: CancellationToken)
 		});
 	});
 
-	return new class implements CancelablePromise<T> {
+	return <CancelablePromise<T>>new class implements Omit<CancelablePromise<T>, typeof Symbol.toStringTag> {
```
Comment (mjbvz) [inline src/vs/base/common/async.ts]: Not sure if there's a better way to express this. The problem is that this file needs to work with our ie11 monaco build as well, which doesn't include `Symbol`

**Comment by mjbvz** [comment]
@joaomoreno Could use your help with the errors related to `es6-promise.d.ts`

**Comment by joaomoreno** [comment]
@jrieken Should jump in. :+1:

**Comment by mjbvz** [comment]
@jrieken I need help with `es6-promise.d.ts` so Let's try to fix this in December for debt week

**Review by bpasero — APPROVED** [review]
Awesome 👍

**Pull request merged**